### PR TITLE
solution

### PR DESCRIPTION
**Pull request by mykola-koval**
[DEMO LINK](https://mykola-koval.github.io/react_movies-list-fetch-movies/)

### CODE REVIEW

**Comment by YevheniiKorzun** [comment]
> [DEMO LINK](https://mykola-koval.github.io/react_movies-list-fetch-movies/)

The form adds null to the list if the movie is not found.

File: src/components/FindMovie/FindMovie.tsx [hunk 1]
```diff
@@ -1,54 +1,115 @@
...
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+export const FindMovie: React.FC<Props> = ({ movies, setMovies }) => {
+  const [movieTitle, setMovieTitle] = useState('');
+  const [movie, setMovie] = useState({
+    Poster: '', Title: '', Plot: '', imdbID: '',
+  });
```
Comment (YevheniiKorzun) [inline src/components/FindMovie/FindMovie.tsx:16]: Don't create a movie! Use server to get it!

File: src/components/FindMovie/FindMovie.tsx [hunk 2]
```diff
@@ -1,54 +1,115 @@
...
+
+  const pushMovie = () => {
+    const movieIsUnique = !(movies.some(mov => mov.imdbID === movie.imdbID));
+
+    if (movieIsUnique) {
+      setMovies([...movies, movie]);
+    } else {
+      setListError(true);
+    }
+  };
```
Comment (YevheniiKorzun) [inline src/components/FindMovie/FindMovie.tsx:40]: Check if the movie was downloaded from the server before submitting.

**Review by YevheniiKorzun — CHANGES_REQUESTED** [review]

**Review by YevheniiKorzun — APPROVED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,29 +1,22 @@
...
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 
-interface State {
-  movies: Movie[];
-}
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState<Movie[] | never[]>([]);
```
Comment (Denchepornyuk) [inline src/App.tsx:7]:
```suggestion
  const [movies, setMovies] = useState<Movie[] | []>([]);
```

File: src/components/FindMovie/FindMovie.tsx [hunk 4]
```diff
@@ -1,54 +1,116 @@
...
+      setSearchError(true);
+      setMovie(null);
+    } else {
+      setMovie(movieFromServer);
+    }
+  };
+
+  const pushMovie = () => {
+    if (movie) {
+      const movieIsUnique = !(movies.some(mov => mov.imdbID === movie.imdbID));
```
Comment (Denchepornyuk) [inline src/components/FindMovie/FindMovie.tsx:33]: use method `find` if you want to find one element in array

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**